chain/events/filter/index.go

```diff
@@ -627,20 +773,29 @@ func (ei *EventIndex) prefillFilter(ctx context.Context, f *eventFilter, exclude
 		}
 	}
 
-	if excludeReverted {
-		clauses = append(clauses, "event.reverted=?")
-		values = append(values, false)
-	}
-
-	if len(f.addresses) > 0 {
-		subclauses := make([]string, 0, len(f.addresses))
-		for _, addr := range f.addresses {
-			subclauses = append(subclauses, "emitter_addr=?")
-			values = append(values, addr.Bytes())
+	// Resolve emitters from filter addresses at the latest TipSet.
+	// TODO investigate if it is safe to use the current tipset; what about finality?
+	emitters, hasAddrs := f.emitters(ctx, nil)
+	if hasAddrs {
+		// The filter has at least one address.
+		if len(emitters) == 0 {
+			// Length of emitters is zero, meaning none of the addresses in the event filter
+			// were resolvable, and therefore there is no event that would match the fillter.
+			return nil
+		}
+		subclauses := make([]string, 0, len(emitters))
+		for emitter := range emitters {
+			subclauses = append(subclauses, "emitter=?")
+			values = append(values, emitter)
 		}
 		clauses = append(clauses, "("+strings.Join(subclauses, " OR ")+")")
 	}
 
+	// Always exclude reverted events when prefilling. See:
+	// - https://github.com/filecoin-project/lotus/issues/11770
+	clauses = append(clauses, "event.reverted=?")
```

Comment:
I disagree with this as an outcome to 11770, see https://github.com/filecoin-project/lotus/issues/11770#issuecomment-2015678344

I think we have to leave this out for now and resolve it separately. I like the idea of an API where you can say "this is where I was last, catch me up". But we don't have that yet so the best we can do is get everything from the API and build a state representation locally by consuming reverted events to undo existing state we have already collected.

```suggestion
```